std/file.d

```diff
@@ -3200,6 +3200,219 @@ version(Posix) unittest //issue 11434
     assert(getAttributes(t2) == octal!100767);
 }
 
+/***************************************************
+Copy file or directory $(D from) to $(D to). When a directory is copied, its
+contents are copied as well, recursively.
+
+File timestamps are preserved, except for symbolic links.
+File attributes are preserved, if $(D preserve) equals $(D PreserveAttributes.yes).
+On Windows only $(D PreserveAttributes.yes) (the default on Windows) is supported.
+
+Existing target files are overwritten. Existing target directories are written
+into.
+
+With symbolic links, the links themselves are copied. That is, for every
+symbolic link in $(D from), a new link with the same target path is created in
+$(D to). Relative target paths are not changed. They may be invalidated by the
+copying, when they point outside of the copied directory tree.
+
+Special files (e.g. device files, FIFOs) cannot be copied with $(D copyRecurse).
+
+Params:
+    from = string or range of characters representing the existing
+        file/directory name
+    to = string or range of characters representing the target file/directory
+        name
+    preserve = whether to preserve the file attributes
+
+Throws: $(D FileException) on error.
+ */
+void copyRecurse(string from, string to,
```

Comment:
Why would recursive be optional? Just use `copy` if that's what you want. Copying a directory is recursive by its very nature; otherwise, you'd just be doing mkdir.